### PR TITLE
parser, expression: follow mysql, increase interval precedence

### LINKED ISSUE #30519: Interval() with expressions is not working in Tidb
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
```sql
SET sql_mode = 'NO_ENGINE_SUBSTITUTION';
SELECT (INTERVAL(1,0,1,2) + 5) * 7 + INTERVAL(1,0,1,2) / 2;
SELECT INTERVAL(0,(1*5)/2) + INTERVAL(5,4,3);
```

### 2. What did you expect to see? (Required)
```sql
mysql> SELECT (INTERVAL(1,0,1,2) + 5) * 7 + INTERVAL(1,0,1,2) / 2;
+-----------------------------------------------------+
| (INTERVAL(1,0,1,2) + 5) * 7 + INTERVAL(1,0,1,2) / 2 |
+-----------------------------------------------------+
|                                             50.0000 |
+-----------------------------------------------------+
1 row in set (0.00 sec)

mysql> SELECT INTERVAL(0,(1*5)/2) + INTERVAL(5,4,3);
+---------------------------------------+
| INTERVAL(0,(1*5)/2) + INTERVAL(5,4,3) |
+---------------------------------------+
|                                     2 |
+---------------------------------------+
1 row in set (0.00 sec)
```
### 3. What did you see instead (Required)
```sql
mysql> SELECT (INTERVAL(1,0,1,2) + 5) * 7 + INTERVAL(1,0,1,2) / 2;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use line 1 column 58 near "" 
mysql> SELECT INTERVAL(0,(1*5)/2) + INTERVAL(5,4,3);
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use line 1 column 44 near "" 
```
### 4. What is your TiDB version? (Required)
```sql
| Release Version: v5.4.0-alpha-264-g6efa36df6
Edition: Community
Git Commit Hash: 6efa36df6cff325106f67ecfe3d79816ba37ca6a
Git Branch: master
UTC Build Time: 2021-11-29 16:57:51
GoVersion: go1.17.2
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false |
```

### PR DESCRIPTION
**Pull request by Defined2014**
### What problem does this PR solve?

Issue Number: close #30519

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
raise `interval` symbol precedence.
```

### CODE REVIEW

File: parser/parser.y [hunk 1]
```diff
@@ -6920,7 +6919,7 @@ FunctionNameConflict:
 |	"DAY"
 |	"HOUR"
 |	"IF"
-|	"INTERVAL" %prec lowerThanIntervalKeyword
```
Comment (Defined2014) [inline parser/parser.y:6923]: Don't know why we set %prec before.
Follow the  [MySQL](https://github.com/mysql/mysql-server/blob/3290a66c89eb1625a7058e0ef732432b6952b435/sql/sql_yacc.yy#L1416) to fix it.

**Comment by Defined2014** [comment]
/cc @xhebox @bb7133 @morgo 
PTAL

File: parser/parser.y (hunk 1 above)
Comment (xhebox) [reply parser/parser.y:6923]: FYI, https://github.com/pingcap/tidb/pull/2370, this is where it is introduced.

**Review by xhebox — APPROVED** [review]

**Comment by Defined2014** [comment]
/cc @xiongjiwei 
PTAL

**Review by xiongjiwei — APPROVED** [review]

**Comment by Defined2014** [comment]
@xiongjiwei @xhebox 
Cloud you help me merge it?

**Pull request merged**